src/document/json/rht_pq_map.rs

```diff
@@ -0,0 +1,255 @@
+use crate::document::json::element::Element;
+use crate::document::time::ticket::Ticket;
+
+use std::cmp::Ordering;
+use std::collections::{BinaryHeap, HashMap};
+
+use thiserror::Error;
+
+type BoxedElement = Box<dyn Element>;
+
+impl Clone for BoxedElement {
+    fn clone(&self) -> BoxedElement {
+        self.deepcopy()
+    }
+}
+
+#[derive(Debug, Error)]
+enum RHTPQMapError {
+    #[error("fail to find : {0}")]
+    ElementNotFound(String),
+}
+
+struct RHTPQMapNode {
+    key: String,
+    element: BoxedElement,
+}
+
+impl Ord for RHTPQMapNode {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.element.created_at().after(&other.element.created_at()) {
+            true => Ordering::Greater,
+            _ => Ordering::Less,
+        }
+    }
+}
+
+impl PartialOrd for RHTPQMapNode {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for RHTPQMapNode {
+    fn eq(&self, other: &Self) -> bool {
+        self.key == other.key
+    }
+}
+
+impl Eq for RHTPQMapNode {}
+
+impl Clone for RHTPQMapNode {
+    fn clone(&self) -> RHTPQMapNode {
+        RHTPQMapNode {
+            key: self.key.clone(),
+            element: self.element.deepcopy(),
+        }
+    }
+}
+
+impl RHTPQMapNode {
+    pub fn new(key: String, element: BoxedElement) -> RHTPQMapNode {
+        RHTPQMapNode { key, element }
+    }
+
+    pub fn remove(&self, ticket: Ticket) -> bool {
+        return self.element.remove(ticket);
+    }
+
+    pub fn is_removed(&self) -> bool {
+        match self.element.removed_at() {
+            Some(_) => true,
+            _ => false,
+        }
+    }
+
+    pub fn key(&self) -> String {
+        self.key.clone()
+    }
+}
+
+pub struct RHTPriorityQueueMap {
+    node_queue_map_by_key: HashMap<String, BinaryHeap<RHTPQMapNode>>,
+    node_map_by_created_at: HashMap<String, RHTPQMapNode>,
```

Comment:
```suggestion
    node_map_by_created_at: HashMap<Ticket, RHTPQMapNode>,
```

In https://github.com/yorkie-team/yorkie-rust-sdk/pull/12, we made the Ticket usable as a hashmap key. I think it would be better to use Ticket as the key type instead of String, what do you think?